### ISSUE TITLE
Icons not showing for loot and roll frames

### ISSUE THREAD

**Issue by stinkybagel** [Xuerian/XLoot#17]
Icons do not show correctly on either the loot frame or the roll frame. It instead shows a black box.
![wow-loot-frame-issue](https://user-images.githubusercontent.com/124319748/216481938-794ea732-aa86-4272-a3a6-761c103bd822.jpg)

**Comment by Xuerian** [comment]
Thanks for reporting this. I'm not sure what's causing it.

It looks like you're on classic, correct?

It's been reported here: https://www.wowace.com/projects/xloot/issues/212

But just like there, I don't have access to classic, and without errors or ideas I don't know what's going on. A cursory googling didn't show anything recent with [GetLootSlotInfo](https://wowpedia.fandom.com/wiki/API_GetLootSlotInfo) not returning a working icon or anything related, and it seems to work on live, correct?

**Comment by stinkybagel** [comment]
Hello, thanks for the quick reply. Yes this is happening on Classic Era/SoM, game version 1.14.3. It seems to have happened with the recent WoW classic WoTLK Patch for Phase 2 Ulduar? That is when I noticed it has stopped working.

Oddly enough, I do not see this issue on WoW Classic WoTLK 3.4.1.

I am happy to try and run a debugger or some way to capture the errors. I am not too familiar with the preferred options to do so though.

**Comment by Xuerian** [comment]
I have no idea what would be causing it. Thanks for trying to bisect.

The only thing I could suggest is opening `XLoot Frame\Frame.lua` and finding 
`		self.texture_item:SetTexture(slotData.icon)`
and adding
`		xprint("%s texture: %s":format(slotData.name, slotData.icon))`
before it.

Then /reload and show me what it says when it shows you blanks.

**Comment by stinkybagel** [comment]
Maybe I am dumb but do you mean like this? Because both of these throw a LUA error due to syntax.

Message:  Interface\AddOns\XLoot_Frame\Frame.lua:693: ')' expected near ':'
Time: Thu Feb  2 18:07:56 2023
Count: 2

--------------------

```
692 -- Icon
693 xprint("%s texture: %s":format(slotData.name, slotData.icon))
694 self.texture_item:SetTexture(slotData.icon)
```

OR

```
692 -- Icon
693 xprint("%s texture: %s":format(slotData.name, slotData.icon))self.texture_item:SetTexture(slotData.icon)
```

**Comment by Xuerian** [comment]
The first one, but I gave you a typo anyway.
`xprint(("%s texture: %s"):format(slotData.name, slotData.icon))`
should work

**Comment by stinkybagel** [comment]
Hello, that worked, thanks!

So I see no visual change in behavior.

Chat printed:  
XLoot: 35 Copper texture: 133769
XLoot: Murloc Fin texture: 134304
![wow-loot-frame-issue-after-change](https://user-images.githubusercontent.com/124319748/216488440-c5c4e9a1-2383-453b-a5ec-aba95d7d2b0e.jpg)

**Comment by Xuerian** [comment]
It's the correct [FileDataID](https://wowpedia.fandom.com/wiki/FileDataID) as shown [here](https://wow.tools/files/#search=133789&page=1&sort=0&desc=asc)

I don't know why it wouldn't work 😕 Thanks for checking. You can remove the line, not that you'd really want to use it while it's not showing you any icons anyway.

Given it works in the other versions, my only guess is it's some obtuse new wow bug. It sets the texture the same way the vanilla loot frame does.

**Comment by stinkybagel** [comment]
No worries, thank you also for trying some quick things. I saw a commenter on WoWAce mentioned the latest version of 9-xx was working, so I will try that as well.

**Comment by stinkybagel** [comment]
I can confirm that the 9.2.7-1 release is working.
![9271-error-working](https://user-images.githubusercontent.com/124319748/216491745-ef23fffd-b8d4-4bc6-b5bc-73f1d3723887.jpg)

**Comment by Xuerian** [comment]
Interesting. Would you be willing to add the same line to that version and show me what it says? It should be the same, but I'd like to verify.

Also, can you confirm this is on an official classic server?

**Comment by stinkybagel** [comment]
Yes this server I am testing from is Shadowstrike, under the Season of Mastery Tab on WoW Classic from USA. I have another character on Whitemane, under the Classic Era tab from USA. Both exhibit the same issue with the 10-xx version.

**Comment by stinkybagel** [comment]
PFA print results in picture to confirm. Using 9.2.7-1 version, adding print line.

XLoot: 54 Copper texture: 133789
XLoot: Thick-shelled Clam texture: 134432
![wow-loot-frame-issue--NEW-PRINT-97](https://user-images.githubusercontent.com/124319748/216496614-e9da0ac5-e967-41a8-99d6-35c33d3302c7.jpg)

**Comment by stinkybagel** [comment]
I also noticed the frame itself in 10-xx version is darker, is there a chance that something is obfuscating the window visually, rather than it being an icon/id failing to be pulled?

**Comment by Xuerian** [comment]
> I also noticed the frame itself in 10-xx version is darker, is there a chance that something is obfuscating the window visually, rather than it being an icon/id failing to be pulled?

Would you be able to confirm if 10-x has the same black background/no gradient visual on other game versions for you?

**Comment by stinkybagel** [comment]
At the moment I do not have retail installed, but I was able to check WoW Classic WoTLK 3.4.1 on server Whitemane (US) using 10-xx, and it doesn't have the dark background/gradient issue.
![wow-loot-frame-issue-wotlk-10x](https://user-images.githubusercontent.com/124319748/216517355-1ca0cce9-9480-4dd2-901a-e91a12e7afff.jpg)

**Comment by Xuerian** [comment]
I appreciate you checking those for me.

**Comment by stinkybagel** [comment]
No problem. Let me know if you need me to test anything else. For now, I'm using 9.2.7-1 on Classic Era content. Thanks
